cluster-autoscaler/provisioningrequest/checkcapacity/condition_test.go

```diff
@@ -0,0 +1,236 @@
+			prConditions: []v1.Condition{
+				{
+					Type:   v1beta1.CapacityFound,
+					Status: v1.ConditionTrue,
+				},
+				{
+					Type:   v1beta1.Provisioned,
+					Status: v1.ConditionTrue,
+				},
+			},
+			want: true,
+		},
+		{
+			name: "Capacity is not found",
+			prConditions: []v1.Condition{
+				{
+					Type:   v1beta1.CapacityFound,
+					Status: v1.ConditionFalse,
+				},
+			},
+			want: false,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			pr := provreqwrapper.NewV1Beta1ProvisioningRequest(
+				&v1beta1.ProvisioningRequest{
+					Spec: v1beta1.ProvisioningRequestSpec{
+						ProvisioningClassName: v1beta1.ProvisioningClassCheckCapacity,
+					},
+					Status: v1beta1.ProvisioningRequestStatus{
+						Conditions: test.prConditions,
+					},
+				}, nil)
+			got := shouldCapacityBeBooked(pr)
+			if got != test.want {
+				t.Errorf("Want: %v, got: %v", test.want, got)
+			}
+		})
+	}
+}
+
+func TestSetCondition(t *testing.T) {
+	tests := []struct {
+		name          string
+		oldConditions []v1.Condition
+		newType       string
+		newStatus     v1.ConditionStatus
+		want          []v1.Condition
+	}{
+		{
+			name:      "CapacityFound added, empty conditions before",
+			newType:   v1beta1.CapacityFound,
+			newStatus: v1.ConditionTrue,
+			want: []v1.Condition{
+				{
+					Type:   v1beta1.CapacityFound,
+					Status: v1.ConditionTrue,
+				},
+			},
+		},
+		{
+			name: "CapacityFound updated",
+			oldConditions: []v1.Condition{
+				{
+					Type:   v1beta1.CapacityFound,
+					Status: v1.ConditionFalse,
+				},
+			},
+			newType:   v1beta1.CapacityFound,
+			newStatus: v1.ConditionTrue,
+			want: []v1.Condition{
+				{
+					Type:   v1beta1.CapacityFound,
+					Status: v1.ConditionTrue,
+				},
+			},
+		},
+		{
+			name: "Failed added, non-empty conditions before",
+			oldConditions: []v1.Condition{
+				{
+					Type:   v1beta1.CapacityFound,
+					Status: v1.ConditionTrue,
+				},
+			},
+			newType:   v1beta1.Failed,
+			newStatus: v1.ConditionTrue,
+			want: []v1.Condition{
+				{
+					Type:   v1beta1.CapacityFound,
+					Status: v1.ConditionTrue,
+				},
+				{
+					Type:   v1beta1.Failed,
+					Status: v1.ConditionTrue,
+				},
+			},
+		},
+		{
+			name: "Unknown type, conditions are not updated",
+			oldConditions: []v1.Condition{
+				{
+					Type:   v1beta1.CapacityFound,
+					Status: v1.ConditionTrue,
+				},
+			},
+			newType:   v1beta1.Provisioned,
+			newStatus: v1.ConditionFalse,
+			want: []v1.Condition{
+				{
+					Type:   v1beta1.CapacityFound,
+					Status: v1.ConditionTrue,
+				},
+			},
+		},
+		{
+			name:      "Expired, empty conditions before",
+			newType:   v1beta1.Expired,
+			newStatus: v1.ConditionFalse,
+			want: []v1.Condition{
+				{
+					Type:   v1beta1.Expired,
+					Status: v1.ConditionFalse,
+				},
+			},
+		},
+		{
+			name: "Capacity found with unknown condition before",
+			oldConditions: []v1.Condition{
+				{
+					Type:   v1beta1.Provisioned,
```

Comment:
`Provisioned` is not strictly unknown, can we put some string there like `UnknownConditionType` ?